cmdb/framework/cmdb_object_manager.py

```diff
@@ -357,7 +357,7 @@ def get_object_references(self, public_id: int, active_flag=None, user: UserMode
 
         return referenced_by_objects
 
-    def delete_object(self, public_id: int, user: UserModel, permission: AccessControlPermission):
+    def delete_object(self, public_id: int, user: UserModel, permission: AccessControlPermission = None):
```

Comment:
`User` should also default to None.
This makes no difference for the access validation. However, it may be that the `delete_object` signature must be changed in many places.

Reply:
When deleting objects, the user is required to generate Cmdb events. If no user is passed, the logic must be adapted accordingly. Either no event or event without user. The user is required since 2019. Only the permissions have a default value, so no further changes are necessary.

Reply:
Then we have a problem with the workflow here. Is then however no part of your work.